homeassistant/components/melnor/__init__.py

```diff
@@ -0,0 +1,56 @@
+"""The melnor integration."""
+
+
+from melnor_bluetooth.device import Device
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DEFAULT_CONNECTION_TIMEOUT, DOMAIN
+from .models import MelnorDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up melnor from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {}).setdefault(entry.entry_id, {})
+
+    ble_device = bluetooth.async_ble_device_from_address(hass, entry.data[CONF_ADDRESS])
```

Comment:
The `ble_device` can change. You'll need a way to feed new `BLEDevice` objects to the library when they change
https://developers.home-assistant.io/docs/network_discovery?_highlight=bluet#best-practices-for-library-authors